flexmeasures/ui/templates/crud/asset.html

```diff
@@ -369,6 +483,436 @@ <h3>All child assets for {{ asset.name }}</h3>
+                        : `<button class="btn btn-primary btn-sm" onclick="addSensorAsGraph(${sensor.id})">Add to Graph</button>`
+                        }
+                        
+                    </div>
+                </div>
+            `;
+
+            apiSensorsList.appendChild(col);
+        });
+    }
+
+    async function updateAssetSensorsToShow() {
+        const graphSpinnerElement = document.getElementById('spinner');
+        // Show spinner
+        graphSpinnerElement.style.display = 'block';
+
+        const apiURL = apiBasePath + "/api/v3_0/assets/{{ asset.id }}";
+        const dataJson = JSON.stringify( {sensors_to_show: JSON.stringify(data)} );
+    
+        const response = await fetch(apiURL, {
+            method: "PATCH",
+            headers: {
+                "Content-Type": "application/json",
+            },
+            body: dataJson,
+        });
+    
+        if (!response.ok) {
+            showToast("Failed to update the asset", "error");
+        } else{
+            document.dispatchEvent(new Event('sensorsToShowUpdated'));
+            showToast("Changes saved successfully", "success");
+        }
+    }
+
+    // Add a sensor as a new graph card
+    function addSensorAsGraph(id) {
+        const newAsset = {
+            title: "No Title", 
+            sensors: [id], 
+        };
+        data.push(newAsset);
+        renderGraphCards();
+    }
+
+    // Add blank graph to the graph cards
+    function addBlankGraph() {
+        const newAsset = {
+            title: "No Title " + (data.length + 1), 
+            sensors: [], 
+        };
+        selectedGraphTitle = newAsset.title;
+        data.push(newAsset);
+        selectGraph(data.length - 1)
+        renderGraphCards();
+    }
+
+    // Add Sensor to an existing graph card
+    function addSensorToExistingGraph(graphIndex, sensorId) {
+        data[graphIndex].sensors.push(sensorId);
+        renderGraphCards();
+    }
+
+    // Remove sensor from the graph sensor list
+    function removeSensorFromGraph(graphIndex, sensorIndex) {
+        data[graphIndex].sensors.splice(sensorIndex, 1);
+        selectedGraphTitle = data[graphIndex].title;
+        renderGraphCards();
+    }
+
+    function handleEnterKeyEvent(event, index) {
+        if (event.key === "Enter") {
+            saveGraphTitle(index);
+        }
+    }
+    // Initial render of graph cards
+    renderGraphCards();
+
+    formModal.addEventListener('hidden.bs.modal', function () {
+        updateAssetSensorsToShow();
+    });
+
+    document.addEventListener("click", function(event) {
+        const card = event.target.closest(".card-highlight");
+        const sensorCard = event.target.closest(".sensor-card");
+        const searchInput = event.target.id === "searchInput";
+        const cardBody = event.target.closest(".card-body");
+
+        
```

Comment:
Please add a comment here (or docstring for the function), explaining your intentions. I see two things happening:

- toggle highlighted card status
- reload sensor list if anything was clicked that is not a card, sensor or search field.

Actually, that last thing is maybe what I would remove. I've said this a few times - removing the sensor search results is not needed unless we re-do the search.
And I'm getting weird behavior, e.g. the sensor results disappear when I am removing a sensor from the graph, probably due to this black-box catch-all `else`.

Reply:
I'll add a comment here to better explain whats happening here.

Majorly this is to handle the clicks on a card and it doesn't do any sensor reload, the issue you where facing is probably a bug, but I have done a couple of refactoring so you can try again to see if you are still having that issue.

Also you can see the newly added comment on what this black does.